AddOns/BasicFinancials/app/src/codeunit/AssistedSetupBF.Codeunit.al

```diff
@@ -0,0 +1,46 @@
+codeunit 20605 "Assisted Setup BF"
+{
+    Access = Internal;
+
+    var
+        AssistedSetupTxt: Label 'Set up the Business Central Basic extension';
+        AssistedSetupHelpTxt: Label 'https://go.microsoft.com/fwlink/?linkid=', Locked = true;
+        AssistedSetupDescriptionTxt: Label 'Business Central Basic';
+        AlreadySetUpQst: Label 'This assisted setup guide has already been completed. Do you want to run it again?';
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Assisted Setup", 'OnRegister', '', false, false)]
+    local procedure SetupInitialize()
+    var
+        BaseAppID: Codeunit "BaseApp ID";
+        AssistedSetup: Codeunit "Assisted Setup";
+        AssistedSetupGroup: Enum "Assisted Setup Group";
+        VideoCategory: Enum "Video Category";
+    begin
+        AssistedSetup.Add(BaseAppID.Get(), PAGE::"Assisted Setup BF", AssistedSetupTxt, AssistedSetupGroup::ReadyForBusiness, '', VideoCategory::ReadyForBusiness, AssistedSetupHelpTxt, AssistedSetupDescriptionTxt);
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Assisted Setup", 'OnReRunOfCompletedSetup', '', false, false)]
+    local procedure OnReRunOfCompletedSetup(ExtensionId: Guid; PageID: Integer; var Handled: Boolean)
+    begin
+        if ExtensionId <> GetAppId() then
+            exit;
+        case PageID of
+            Page::"Assisted Company Setup Wizard":
+                begin
+                    if Confirm(AlreadySetUpQst, true) then
```

Comment:
Do we have test to rerun the assisted setup ? I think it would be valuable to have it.

Reply:
There is no test to rerun the assisted setup. That is primarily because there is no test the run the assisted setup (no test completion of the assisted setup). 
The assisted setup in the current solution cannot be completed (Step 7) in a newly created  Business Central, because of the requirement of Supported Licenses and Supported Companies (Step 2,3,4 needs to be performed)

To install the extension, the partner does the following:
    1.	Creates a new Business Central tenant.
    2.	Adds at least one user who has a Business Central Basic license assigned to them in the Azure Active Directory tenant.
    3.	Removes all companies manually, including Cronus.
    4.	Creates one production company that does not contain data or setups.
    5.	Imports a configuration package that includes Basic setup data.
    6.	Installs the Basic extension. During installation the extension verifies the country availability that there is only one company.
    7.	Completes the Business Central Basic assisted setup guide. The assisted setup guide checks for the Business Central Basic license.
    8.	Sends sign in information to the customer in an email.